sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -655,8 +678,21 @@ def test_check_estimator():
         "Estimator LargeSparseNotSupportedClassifier doesn't seem to "
         r"support \S{3}_64 matrix, and is not failing gracefully.*"
     )
-    with raises(AssertionError, match=msg):
-        check_estimator(LargeSparseNotSupportedClassifier())
+    with warnings.catch_warnings(record=True) as records:
+        warnings.filterwarnings(
+            "always",
+            category=DeprecationWarning,
+            message=(
+                "`getformat` is deprecated and will be removed in v1.13.0; use"
+                " `X.format` instead."
+            ),
+        )
```

Comment:
I'm a bit confused by this.

Reply:
@adrinjalali 
scipy has deprecated `getformat` among other methods for sparse arrays with version 1.11.0 ([see here](https://scipy.github.io/devdocs/release/1.11.0-notes.html)). Despite using version 1.11.3 myself, I don't get this warning locally, it only popped up in the CI...

If I remember correctly, some tests had failed because of the warning. Should they have been filtered out anyways? Or what's the reason for the confusion?

Reply:
Then we should use `X.format` as suggested by the warning I think.

Reply:
Okay, @adrinjalali, I have just swapped that. Please have another look.